examples/fcu_custom_type.cpp

```diff
@@ -2,13 +2,16 @@
 
 #include <iostream>
 
-struct MyIdent : public msp::Request {
+struct MyIdent : public msp::Message {
+    MyIdent(msp::FirmwareVariant v) : Message(v) {}
+    
     msp::ID id() const { return msp::ID::MSP_IDENT; }
 
     msp::ByteVector raw_data;
 
-    void decode(const msp::ByteVector &data) {
+    bool decode(msp::ByteVector &data) {
```

Comment:
Its supposed be be a virtual override method, so the function signature needs to match. Normally, the decode method does manipulate the internal state of the ByteVector. I suppose I could declare the relevant parts mutable, making it logically const-correct instead of just bitwise const.